### PR TITLE
MBS-8971: Add intro

### PR DESCRIPTION
**Pull request by sh-csg**

### CODE REVIEW

File: version.php [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
  * @author      Stefan Hanauska <stefan.hanauska@csg-in.de>
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->component = 'mod_learningmap';
-$plugin->release = '0.9.7';
-$plugin->version = 2024022102;
+$plugin->release = '0.9.8';
+$plugin->version = 2024032402;
```
Comment (TobiGa) [inline version.php:29]: Because of the changes in the meantime, the version needs to be updated.

File: styles.css [hunk 2]
```diff
@@ -94,6 +94,10 @@
     position: relative;
 }
 
+#page-mod-learningmap-mod #fitemid_id_svgcode {
+    display: none;
```
Comment (TobiGa) [inline styles.css:98]: Typo: fitem_id_svgcode instead of fitemid_id_svgcode.

File: backup/moodle2/restore_learningmap_activity_task.class.php [hunk 3]
```diff
@@ -99,25 +100,29 @@ public function after_restore(): void {
...
             }
-            $mapworker = new \mod_learningmap\mapworker($item->intro, (array)$placestore);
+            if (empty($item->svgcode)) {
+                $mapcode = $item->intro;
+                $item->intro = '';
+                $item->showmaponcoursepage = $cm->showdescription;
+            } else {
+                $mapcode = $item->svgcode;
+            }
+            $mapworker = new \mod_learningmap\mapworker($mapcode, (array)$placestore);
```
Comment (TobiGa) [inline backup/moodle2/restore_learningmap_activity_task.class.php:118]: When you backup in the old version and restore in the new, the filearea is "intro" and not "background".

**Comment by sh-csg** [comment]
Rebased to current master, fixed all points mentioned

**Pull request merged**